workflow-steps/cache/hashing-utils.ts

```diff
@@ -40,5 +36,14 @@ export function hashKey(key: string): string {
   const globHashes = globsToHash.map((globPattern) => {
     return hashFileContents(globPattern);
   });
-  return [...hardcodedKeys, ...globHashes].join(' | ');
+  const hashCollections = [...hardcodedKeys, ...globHashes];
+
+  if (hashCollections.length > 1) {
+    return hash(hashCollections.join(' | '));
+  }
+  return hashCollections.join(' | ');
```

Comment:
I could've just always returned 
```typescript
return hash(hashCollections.join(' | '));
```

but I kept the old way of hashing when it's a single-key for backward compatibility purposes - so we don't bust everyone's cache when this gets merged in

Reply:
I'd leave that in a comment in the code?